token/hmac/hmacsha_test.go

```diff
@@ -133,3 +133,33 @@ func TestCustomHMAC(t *testing.T) {
 	require.NoError(t, sha512.Validate(context.Background(), token512))
 	require.EqualError(t, def.Validate(context.Background(), token512), fosite.ErrTokenSignatureMismatch.Error())
 }
+
+func TestGenerateFromString(t *testing.T) {
+	cg := HMACStrategy{Config: &fosite.Config{
+		GlobalSecret: []byte("1234567890123456789012345678901234567890")},
+	}
+	for _, c := range []struct {
+		text string
+		hash string
+	}{
+		{
+			text: "",
+			hash: "-n7EqD-bXkY3yYMH-ctEAGV8XLkU7Y6Bo6pbyT1agGA=",
```

Comment:
`=` is sometimes a problem in URLs as it collides with `?foo=bar`. Maybe it would be better to use base64 without padding?

Reply:
Done